test/integration/scheduler/priorities_test.go

```
@@ -0,0 +1,174 @@
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	// Add a label to one of the nodes.
+	labeledNode := nodes[1]
+	labelKey := "kubernetes.io/node-topologyKey"
+	labelValue := "topologyvalue"
+	labels := map[string]string{
+		labelKey: labelValue,
+	}
+	if err = testutils.AddLabelsToNode(context.clientSet, labeledNode.Name, labels); err != nil {
+		t.Fatalf("Cannot add labels to node: %v", err)
+	}
+	if err = waitForNodeLabels(context.clientSet, labeledNode.Name, labels); err != nil {
+		t.Fatalf("Adding labels to node didn't succeed: %v", err)
+	}
+	// Create a pod with node affinity.
+	podName := "pod-with-node-affinity"
+	pod, err := runPausePod(context.clientSet, &pausePodConfig{
+		Name:      podName,
+		Namespace: context.ns.Name,
+		Affinity: &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+					{
+						Preference: v1.NodeSelectorTerm{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      labelKey,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{labelValue},
+								},
+							},
+						},
+						Weight: 20,
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatalf("Error running pause pod: %v", err)
+	}
+	if pod.Spec.NodeName != labeledNode.Name {
+		t.Errorf("Pod %v got scheduled on an unexpected node: %v. Expected node: %v.", podName, pod.Spec.NodeName, labeledNode.Name)
+	} else {
+		t.Logf("Pod %v got successfully scheduled on node %v.", podName, pod.Spec.NodeName)
+	}
+}
+
+// TestPodAffinity verifies that scheduler's pod affinity priority function
+// works correctly.
+func TestPodAffinity(t *testing.T) {
+	context := initTest(t, "pod-affinity")
+	defer cleanupTest(t, context)
+	// Add a few nodes.
+	nodesInTopology, err := createNodes(context.clientSet, "in-topology", 5)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	topologyKey := "node-topologykey"
+	topologyValue := "topologyvalue"
```

That would make the test more resilient, but I guess the tests are not supposed to be resilient. Tests are preferred to fail fast when a PR breaks anything in the test or in the integration cluster.